superset-frontend/src/reports/types.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Types mirroring enums in `superset/reports/models.py`:
+ */
+export type ReportScheduleType = 'Alert' | 'Report';
```

Comment:
is appending `Type` to the end of all these types standard? I could see it for `ReportScheduleType` (since it's defining what type of report it is), but seems like we could just use `ReportCreationMethod` and `ReportRecipient` for the other 2

Reply:
I'm using the same variables as in the backend but maybe we can change the backend as well.

Reply:
I think I'll change `ReportCreationMethodType` but keep `ReportRecipientType`, as `ReportRecipient` sounds like _one_ recipient where both a recipient type and relevant metadata have been configured.

Reply:
Hello, I think that I do add these to the backend in this refactor of the report modal. 

https://github.com/apache/superset/pull/19130